packages/protocol/contracts/libs/LibAddress.sol

```diff
@@ -33,14 +35,23 @@ library LibAddress {
     {
         // Check for zero-address transactions
         if (_to == address(0)) revert ETH_TRANSFER_FAILED();
+
+        uint256 gasLimit;
+        if (_gasLimit == 0) {
+            gasLimit = gasleft();
+        } else {
+            // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-150.md
+            if (((gasleft() * 63) >> 6) < _gasLimit) revert NOT_ENOUGH_GASLEFT();
```

Comment:
To be 100% correct, this will require some extra gas to be taken into account for the gas consumed between the `gasleft` call here and the actual gas value left to be sent into the call. So also the Safe contract: https://github.com/safe-global/safe-smart-account/blob/1cd7568769128717c1a6862d22fe34873d7c79c8/contracts/Safe.sol#L160